### ISSUE TITLE
[Feature Request] FnLock HUD

### ISSUE THREAD

**Issue by tylernguyen** [MSzturc/ThinkpadAssistant#25]
Hey @hexart @MSzturc 

I spend quite a bit of my time in Excel so I spam `FnLock` quite often. It would be nice to have a native macOS HUD for this key.

The EC related query is `_Q74`.

I understand this is completely optional, but again, it would be very nice for me to have this feature. And I think it should be a fairly easy implementation apart from deciding the HUD icon.

**Comment by MSzturc** [comment]
You're right the hardest thing would be creating an icon for this. I've searched a bit, and couldn't find an icon at all.

**Comment by Sniki** [comment]
> You're right the hardest thing would be creating an icon for this. I've searched a bit, and couldn't find an icon at all.

What about a lock: locked/unlocked with a keyboard button and FN on it ?

**Comment by hexart** [comment]
OK, I'm in.

**Comment by hexart** [comment]
<img width="623" alt="Fnlock" src="https://user-images.githubusercontent.com/5528016/84871173-c68cb980-b0b2-11ea-9352-da1ea436e8b5.png">

Hey guys, how about this one?

**Comment by hexart** [comment]
<img width="614" alt="fnlock1" src="https://user-images.githubusercontent.com/5528016/84872088-ff795e00-b0b3-11ea-9f15-6e2f296ab23c.png">

I redesigned the two letters to make it look better.

**Comment by tylernguyen** [comment]
@hexart I like the `fn` but I would think the dots should a be a bit bigger to make them more distinct from one another.

@MSzturc Maybe the texts below should be `Function Keys` and `Multimedia Keys`? or perhaps shorter: `Function`/`Media`

**Comment by hexart** [comment]
<img width="657" alt="9 00 28" src="https://user-images.githubusercontent.com/5528016/84965698-2f6f4280-b142-11ea-92a8-174c9de95b75.png">

**Comment by tylernguyen** [comment]
@hexart I like the sizing on the right. What would the empty one look like?

**Comment by hexart** [comment]
<img width="652" alt="9 18 23" src="https://user-images.githubusercontent.com/5528016/85086671-db319480-b20d-11ea-91cf-6f3907cc3679.png">

**Comment by tylernguyen** [comment]
@hexart @MSzturc I think those icons are perfect.

As for text, I leave it to you @MSzturc, I'm thinking about some permutations of:
`Function Keys/ Multimedia Keys`
`Function/ Media`
`Function Keys/ Media Keys`

Can't wait for this to be added!

**Comment by hexart** [comment]
[fnlockoff.pdf](https://github.com/MSzturc/ThinkpadAssistant/files/4802647/fnlockoff.pdf)
[fnlockon.pdf](https://github.com/MSzturc/ThinkpadAssistant/files/4802648/fnlockon.pdf)

**Comment by MSzturc** [comment]
Looks great, i will implement this feature tomorrow.
@tylernguyen May i ask you to write the ACPI Code. I plan to implement the following bindings:

f18+left shift: fkeys (F1-F12)
f18+right shift: media keys

**Comment by tylernguyen** [comment]
@MSzturc Got it and tested in Karabiner-EventViewer. Here it is:

**Make sure you patch rename _Q74 to XQ74**
```
        Name (LED3, Zero)
        Method (_Q74, 0, NotSerialized) // FnLock (Fn + Esc)
        {
            If (_OSI ("Darwin"))
            {
                // Toggle FnLock LED
                If ((LED3 == Zero))
                {
                    // Right Shift + F18
                    Notify (\_SB.PCI0.LPCB.KBD, 0x012A)
                    Notify (\_SB.PCI0.LPCB.KBD, 0x0369)
                    Notify (\_SB.PCI0.LPCB.KBD, 0x01aa)

                    // 0x02 = Enable LED
                    \_SB.PCI0.LPCB.EC.HKEY.MHKQ (0x02)
                    LED3 = One
                }
                Else
                {
                    // Left Shift + F18
                    Notify (\_SB.PCI0.LPCB.KBD, 0x0136)
                    Notify (\_SB.PCI0.LPCB.KBD, 0x0369)
                    Notify (\_SB.PCI0.LPCB.KBD, 0x01b6)

                    // 0x00 = Disable LED
                    \_SB.PCI0.LPCB.EC.HKEY.MHKQ (Zero)
                    LED3 = Zero
                }

            }
            Else
            {
                // Call original _Q74 method.
                \_SB.PCI0.LPCB.EC.XQ74()
            }
```

A few notes here:
Calling it `LED3` since `LED1` is F4 and `LED2` is keyboard backlight.
`_SB.PCI0.LPCB.EC.HKEY.MHKQ` is the `FnLock` LED device.
Similar to the Mute key, I suspect there exists a pitfall where LED state will not sync past a reboot/shutdown.
In BIOS, my `"F1-F12 as Primary Function is set to [Disabled]"`

**Comment by MSzturc** [comment]
Implemented with [v1.8.0](https://github.com/MSzturc/ThinkpadAssistant/releases/tag/v1.8.0)
